simple-facebook-instant-articles.php

```diff
@@ -425,6 +427,35 @@ public function render_images( \DOMDocument &$dom, \DOMXPath &$xpath ) {
 	}
 
 	/**
+	 * Facebook throws a warning for all headings below h2.
+	 *
+	 * Replace with h2s.
+	 *
+	 * @param  \DOMDocument &$dom   DOM object generated for post content.
+	 * @param  \DOMXPath    &$xpath XPATH object generated for post content.
+	 *
+	 * @return void
+	 */
+	public function fix_headings( \DOMDocument &$dom, \DOMXPath &$xpath ) {
+
+		$headings = array( 'h3', 'h4', 'h5', 'h6' );
+
+		foreach ( $headings as $heading_tag ) {
+
+			foreach ( $dom->getElementsByTagName( $heading_tag ) as $node ) {
+
+				$h2 = $dom->createElement( 'h2' );
+
+				while ( $node->childNodes->length > 0 ) {
+					$h2->appendChild( $node->childNodes->item( 0 ) );
```

Comment:
I'm not so good at trying to explain stuff like this!

`getElementsByTagName` returns a `DOMNodeList` object, not a standard array. It implements a lot of standard array functionality, so can do stuff like foreach. But it also updates live, which is why we don't need to remove items when we append elsewhere. This actually works just like jQuery - so if you're not familiar with that, might seem a bit unusual.

However foreach can get a bit tricky once you start trying to remove stuff from within a foreach when dealing with references. This is why I used the while loop, as we're fetching the first item each time and its more robust.

In fact I just tested this again and noticed I've been caught out trying to use a foreach, literally on the line above ;)